application/src/test/java/org/opentripplanner/updater/GtfsRealtimeFuzzyTripMatcherTest.java

```diff
@@ -2,47 +2,131 @@
+  private static final String TRIP_ID = "t1";
+  private static final Trip TRIP = TimetableRepositoryForTest.trip(TRIP_ID).build();
+
+  private static final FeedScopedId SERVICE_ID = TimetableRepositoryForTest.id("sid1");
+  private static final String START_TIME = "07:30:00";
+  private static final RealTimeTripTimes TRIP_TIMES = TripTimesFactory.tripTimes(
+    TRIP,
+    TEST_MODEL.stopTimesEvery5Minutes(5, TRIP, START_TIME),
+    new Deduplicator()
+  );
+  private static final TripPattern TRIP_PATTERN = TimetableRepositoryForTest
+    .tripPattern("tp1", ROUTE)
+    .withStopPattern(TimetableRepositoryForTest.stopPattern(STOP_1, STOP_2))
+    .withScheduledTimeTableBuilder(builder -> builder.addTripTimes(TRIP_TIMES))
+    .build();
 
-    GtfsRealtimeFuzzyTripMatcher matcher = new GtfsRealtimeFuzzyTripMatcher(
-      new DefaultTransitService(timetableRepository)
+  @BeforeAll
+  static void setup() {
+    TRIP_TIMES.setServiceCode(SERVICE_CODE);
+    CalendarServiceData calendarServiceData = new CalendarServiceData();
+    calendarServiceData.putServiceDatesForServiceId(SERVICE_ID, List.of(SERVICE_DATE));
+    TIMETABLE_REPOSITORY.addTripPattern(TRIP_PATTERN.getId(), TRIP_PATTERN);
+    TIMETABLE_REPOSITORY.getServiceCodes().put(SERVICE_ID, SERVICE_CODE);
+    TIMETABLE_REPOSITORY.updateCalendarServiceData(
+      true,
+      calendarServiceData,
+      DataImportIssueStore.NOOP
     );
-    TripDescriptor trip1 = TripDescriptor
+    TIMETABLE_REPOSITORY.index();
+  }
+
+  @Test
+  void noTripId() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().build();
+    assertEquals(TRIP.getId().getId(), matcher.match(FEED_ID, trip1).getTripId());
+  }
+
+  @Test
+  void tripIdSetButNotInSchedule() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setTripId("does-not-exist-in-schedule").build();
+    assertEquals(TRIP_ID, matcher.match(FEED_ID, trip1).getTripId());
+  }
+
+  @Test
+  void tripIdExistsInSchedule() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setTripId(TRIP_ID).build();
+    assertEquals(TRIP_ID, matcher.match(FEED_ID, trip1).getTripId());
+  }
+
+  @Test
+  void incorrectRoute() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setRouteId("does-not-exists").build();
+    assertFalse(matcher.match(FEED_ID, trip1).hasTripId());
+  }
+
+  @Test
+  void incorrectDateFormat() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setStartDate("ZZZ").build();
+    assertFalse(matcher.match(FEED_ID, trip1).hasTripId());
+  }
+
+  @Test
+  void incorrectDirection() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setDirectionId(1).build();
+    assertFalse(matcher.match(FEED_ID, trip1).hasTripId());
+  }
+
+  @Test
+  void noMatch() {
+    // Test matching with "real time", when schedule uses time grater than 24:00
```

Comment:
s/grater/greater

Reply:
I actually meant this: 

![image](https://github.com/user-attachments/assets/a8968998-8f22-4a81-828f-176c8a11c97a)

Reply:
Fixed in f40626194bfc67e2bfdac64f41eb2d69d881d397